pkg/sink/mysql/factory.go

```diff
@@ -16,7 +16,126 @@ package mysql
+// unit test, DBConnectionFactoryForTest allows developers to easily set up custom
+// connection creation methods. The SetTemporaryConnectionFactory and
+// SetStandardConnectionFactory methods allow you to define how the temporary and
+// standard connections should be created during testing.
+//
+// Once these methods are set, the system will automatically invoke CreateTemporaryConnection
+// and CreateStandardConnection to establish the respective connections during the
+// connection creation process. This approach provides flexibility and convenience
+// in unit testing scenarios, where different connection behaviors need to be mocked
+// or tested separately.
+type DBConnectionFactoryForTest struct {
+	temp     ConnectionFactory
+	standard ConnectionFactory
+}
+
+// SetTemporaryConnectionFactory sets the connection factory that will be used to
+// create the temporary connection during testing. This allows for custom behavior
+// during unit tests when different connection logic is required.
+func (d *DBConnectionFactoryForTest) SetTemporaryConnectionFactory(f ConnectionFactory) {
+	d.temp = f
+}
+
+// SetStandardConnectionFactory sets the connection factory that will be used to
+// create the standard connection during testing. This provides the ability to mock
+// the connection behavior specifically for the standard connection phase.
+func (d *DBConnectionFactoryForTest) SetStandardConnectionFactory(f ConnectionFactory) {
+	d.standard = f
+}
+
+// CreateTemporaryConnection creates a temporary database connection during testing
+// using the connection factory set by SetTemporaryConnectionFactory. If no factory
+// has been set, it returns an error. This method allows for customized connection
+// logic during the temporary connection phase in unit tests.
+func (d *DBConnectionFactoryForTest) CreateTemporaryConnection(ctx context.Context, dsnStr string) (*sql.DB, error) {
+	if d.temp == nil {
+		return nil, cerror.ErrCodeNilFunction.GenWithStackByArgs()
+	}
+	return d.temp(ctx, dsnStr)
+}
+
+// CreateStandardConnection creates the standard database connection during testing
+// using the connection factory set by SetStandardConnectionFactory. If no factory
+// has been set, it returns an error. This method supports custom connection behavior
+// during the standard connection phase in unit tests.
+func (d *DBConnectionFactoryForTest) CreateStandardConnection(ctx context.Context, dsnStr string) (*sql.DB, error) {
+	if d.standard == nil {
+		return nil, cerror.ErrCodeNilFunction.GenWithStackByArgs()
+	}
+	return d.standard(ctx, dsnStr)
+}
+
+// ConnectionFactory is a function type that takes a context and a DSN (Data Source Name) string
+// as input parameters, and returns a pointer to an sql.DB object and an error. This function type
+// is typically used to create and configure database connections.
+type ConnectionFactory func(ctx context.Context, dsnStr string) (*sql.DB, error)
```

Comment:
Please move the definition of CreateMySQLDBConn here.

Reply:
Good point! I’ll move the definition of CreateMySQLDBConn as suggested. Thanks for the feedback!